provisioner/ansible/provisioner.go

```diff
@@ -73,14 +74,34 @@ type PassthroughTemplate struct {
 }
 
 func (p *Provisioner) Prepare(raws ...interface{}) error {
+	// This is a bit of a hack. For provisioners that need access to
+	// auto-generated WinRMPasswords, the mechanism of keeping provisioner data
+	// and build data totally segregated breaks down. We get around this by
+	// having the builder stash the WinRMPassword in the state bag, then
+	// grabbing it out of the statebag inside of StepProvision.  Then, when
+	// the time comes to provision for real, we run the prepare step one more
```

Comment:
Is there any danger in re-running the Prepare step?  Do you expect this step to be idempotent?

Reply:
I do expect it to be idempotent; the point of "Prepare" is to assign values from this raw interface into the config and, if the config is empty, to assign a default value to it. When Prepare is run the second time, the config has already been populated, either with user-given values or with default ones, so unless new information is provided in the raw interface, nothing will be overwritten. I did some fairly extensive testing on this and I feel confident that the change is sound, though I do recognize it changes a fundamental assumption so it's a very good question to ask.